include/eve/algo/container/detail/soa_storage2.hpp

```diff
@@ -0,0 +1,188 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Contributors & Maintainers
+  SPDX-License-Identifier: MIT
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/module/core.hpp>
+#include <eve/algo/views/zip.hpp>
+#include <eve/detail/kumi.hpp>
+#include <eve/memory/aligned_allocator.hpp>
+#include <eve/concept/simd_allocator.hpp>
+#include <eve/product_type.hpp>
+#include <type_traits>
+#include <memory>
+
+namespace eve::algo::detail
+{
+  template<eve::product_type Type, eve::simd_allocator Allocator>
+  struct soa_storage : Allocator
+  {
+    using is_product_type = void;
+
+    soa_storage() noexcept
+                : indexes_{}, storage_( nullptr, aligned_deleter{this} ), size_{}, capacity_{}
+    {}
+
+    explicit  soa_storage( std::size_t n) : soa_storage( n, Type{} ) {}
+
+    soa_storage( std::size_t n, Type const& v)  : soa_storage()
+    {
+      storage_t  local{ reinterpret_cast<byte_t*>(Allocator::aligned_alloc(byte_size(n), 64))
+                      , aligned_deleter{this}
+                      };
+      storage_.swap(local);
+      size_     = n;
+      build(n,v);
+    }
+
+    soa_storage(soa_storage const& src) : soa_storage()
+    {
+      auto sz = byte_size(src.size_);
+      storage_t  local{ reinterpret_cast<byte_t*>(Allocator::aligned_alloc(sz,64))
+                      , aligned_deleter{this}
+                      };
```

Comment:
Oh nonono - don't store a pointer to this. Just store allocator in the deleter. There is a [get_deleter](https://en.cppreference.com/w/cpp/memory/unique_ptr/get_deleter) on unique_ptr.

This will also save you pointer.

Reply:
Ok I see 👍